app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates :username, :password, :email, presence: true
+  validates :username, :email, uniqueness: true
+end
```

Comment:
yes, I think we do need their names, probably first and last. good catch @cmacaulay